org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java

```diff
@@ -577,7 +579,9 @@ private static boolean getConvertResolvedTypeToVarTypeProposal(IInvocationContex
 			return false;
 		}
 
-		proposals.add(new TypeChangeCorrectionProposal(context.getCompilationUnit(), varBinding, astRoot, typeBinding, IProposalRelevance.CHANGE_VARIABLE));
+		TypeChangeCorrectionProposal proposal = new TypeChangeCorrectionProposal(context.getCompilationUnit(), varBinding, astRoot, typeBinding, IProposalRelevance.CHANGE_VARIABLE);

```

Comment:
In TypeChangeCorrectionProposal constructor, it passed CodeActionKind.QuickFix as the default kind. Why not change it to the correct kind? Are there other cases reusing it for quick fix?

Reply:
Yes, I searched globally and found the proposal was also used to provide a quickfix when there was a type mismatch. 

And of course an alternative is to add a new constructor for the proposal. I’m ok with either of them.